pkg/api/v1beta3/dynakube/oneagent_props.go

```diff
@@ -77,30 +84,20 @@ func (dk *DynaKube) OneAgentConnectionInfoConfigMapName() string {
 	return dk.Name + OneAgentConnectionInfoConfigMapSuffix
 }
 
-func (dk *DynaKube) NeedsReadOnlyOneAgents() bool {
-	return (dk.HostMonitoringMode() || dk.CloudNativeFullstackMode()) &&
-		dk.FeatureReadOnlyOneAgent()
+func (dk *DynaKube) NeedsCSIDriver() bool {
+	return dk.CloudNativeFullstackMode()
 }
 
-func (dk *DynaKube) IsAppMonitoringWithCSI() bool {
-	if !dk.ApplicationMonitoringMode() {
-		return false
-	}
-
-	defaultUseCSIDriver := false
-	if dk.Spec.OneAgent.ApplicationMonitoring.UseCSIDriver == nil {
-		return defaultUseCSIDriver
-	}
-
-	return *dk.Spec.OneAgent.ApplicationMonitoring.UseCSIDriver
+func (dk *DynaKube) CanUseCSIDriver() bool {
+	return dk.HostMonitoringMode() || dk.ApplicationMonitoringMode()
 }
 
-func (dk *DynaKube) NeedsCSIDriver() bool {
-	isAppMonitoringWithCSI := dk.IsAppMonitoringWithCSI()
-
-	isHostMonitoringWithCSI := dk.HostMonitoringMode() && dk.FeatureReadOnlyOneAgent()
+func (dk *DynaKube) UseCSIDriver() bool {
+	return dk.NeedsCSIDriver() || (dk.CanUseCSIDriver() && dk.isEnabledModules().CSIDriver)
+}
 
-	return dk.CloudNativeFullstackMode() || isAppMonitoringWithCSI || isHostMonitoringWithCSI
+func (dk *DynaKube) NeedsReadOnlyOneAgents() bool {
```

Comment:
Since we are dropping the feature flag, do we still need this check or can it just be a check if CSI driver is used in case of host agent?